src/status_im/ui/screens/profile/ttt/views.cljs

```diff
@@ -0,0 +1,316 @@
+(ns status-im.ui.screens.profile.ttt.views
+  (:require-macros [status-im.utils.views :refer [defview letsubs]])
+  (:require [status-im.ui.components.react :as react]
+            [status-im.ui.components.status-bar.view :as status-bar]
+            [status-im.ui.components.toolbar.view :as toolbar]
+            [status-im.ui.components.toolbar.actions :as actions]
+            [status-im.ui.components.colors :as colors]
+            [status-im.react-native.js-dependencies :as js-dependencies]
+            [status-im.react-native.resources :as resources]
+            [status-im.ui.components.common.common :as components.common]
+            [status-im.utils.money :as money]
+            [re-frame.core :as re-frame]
+            [reagent.core :as reagent]
+            [taoensso.timbre :as log]
+            [status-im.ui.components.text-input.view :as text-input]
+            [status-im.ui.components.icons.vector-icons :as icons]
+            [status-im.ui.components.common.common :as components.common]
+            [status-im.ui.components.common.styles :as components.common.styles]
+            [clojure.string :as string]
+            [status-im.utils.config :as config]
+            [status-im.utils.utils :as utils]
+            [status-im.ui.screens.profile.ttt.styles :as styles]
+            [status-im.i18n :as i18n]
+            [status-im.ui.components.styles :as common.styles]
+            [status-im.utils.platform :as platform]))
+
+(defn separator []
+  [react/view {:style {:flex 1 :height 1 :background-color colors/gray-lighter}}])
+(defn steps-numbers [editing?]
+  {:intro                1
+   :set-snt-amount       (if editing? 1 2)
+   :personalized-message (if editing? 2 3)
+   :finish               3})
+
+(defn step-back [step editing?]
+  (re-frame/dispatch
+   (case step
+     (:intro :edit)                [:navigate-back]
```

Comment:
better to move this logic into dispatch function, and do not use set-in

Reply:
Thanks, good point, moved.